### DISCUSSION TITLE
When the active partition is b0rked ...

### DISCUSSION THREAD

**Discussion by robocoder** [questdb/questdb#2568] [Ideas]
We had a server crash yesterday. When the server was rebooted and questdb restarted, we found the active partition on one of the tables was corrupted.  As a result, all `INSERT`s were failing.

We could generally`SELECT` data prior to the current day but there were statements that would cause the questdb worker to crash. Examples:
```
SELECT * FROM tablename ORDER BY ts DESC LIMIT 1;

INSERT INTO TABLE tablename2 SELECT * FROM tablename;
```
> Note: I didn't keep a copy of the corrupted partition folder but LMK if you're interested in the crash logs.

Since we `...PARTITION BY DAY...`, I thought we'd only have to wait until 00:00 UTC, expecting QuestDB to create a new partition but it didn't. `INSERT`s continued to fail and I could not `ALTER TABLE DROP PARTITION` because the active partition was still the corrupted one.

My restoration process became:
```
# stop processes/cron/supervisor workers that write to table
...

CREATE TABLE tablename2 ( ... );

INSERT INTO TABLE tablename2 SELECT * FROM tablename WHERE ts < '2022-09-29';

DROP TABLE tablename;

RENAME TABLE tablename2 TO tablename;

...
# re-import log data for 2022-09-29

# restart writers
```

----

What's the rationale behind not allowing the active partiton to be detached or dropped, given that there's no similar limitation to drop the entire table? (Being able to drop the active partition could have saved me a few steps.)

**Comment by bluestreak01** [comment]
Sorry to hear of the crash! We're actively working on allowing to drop active partition. Likely next release will have it
